src/github_permission_manager_webhook/tests/test_handler_github_permission_manager.py

```diff
@@ -0,0 +1,244 @@
+
+def test_initialise_aws_clients(github_permission_manager):
+    with patch('boto3.resource') as mock_resource, \
+        patch('boto3.client') as mock_client:
+        github_permission_manager.initialise_aws_clients()
+        mock_resource.assert_called_once_with('dynamodb', region_name='eu-west-2')
+        mock_client.assert_any_call('stepfunctions', region_name='eu-west-2')
+        mock_client.assert_any_call('ssm', region_name='eu-west-2')
+
+def test_get_all_parameters(github_permission_manager):
+    with patch.dict('os.environ', {'WORKSPACE': 'workspace'}), \
+        patch.object(github_permission_manager, 'get_ssm_parameter') as mock_get_ssm_parameter, \
+        patch.object(github_permission_manager, 'get_token_to_access_github') as mock_get_token_to_access_github:
+
+        mock_get_ssm_parameter.side_effect = lambda param: f"mock_{param.split('/')[-1]}"
+        mock_get_token_to_access_github.return_value = {"Authorization": "Bearer mock_token"}
+
+        github_permission_manager.get_all_parameters()
+
+        mock_get_ssm_parameter.assert_any_call('/github_permission_manager_webhook/workspace_app_id')
+        mock_get_ssm_parameter.assert_any_call('/github_permission_manager_webhook/workspace_private_key')
+        mock_get_ssm_parameter.assert_any_call('/github_permission_manager_webhook/workspace_installation_id')
+        mock_get_ssm_parameter.assert_any_call('/github_permission_manager_webhook/workspace_secret_for_webhook')
+        mock_get_ssm_parameter.assert_any_call('/github_permission_manager_demotion/workspace_step_function_arn')
+        # Verify that get_token_to_access_github was called
+        mock_get_token_to_access_github.assert_called_once()
+
+        assert github_permission_manager.app_id == "mock_workspace_app_id"
+        assert github_permission_manager.private_key == "mock_workspace_private_key"
+        assert github_permission_manager.installation_id == "mock_workspace_installation_id"
+        assert github_permission_manager.webhook_secret == "mock_workspace_secret_for_webhook"
+        assert github_permission_manager.step_function_arn == "mock_workspace_step_function_arn"
+        assert github_permission_manager.auth_headers == {"Authorization": "Bearer mock_token"}
+
+def test_post_comment_on_issue(github_permission_manager):
+    payload = {'repository': {'full_name': 'org/repo'}, 'issue': {'number': 1}}
+    comment_body = 'Test comment'
+
+    with patch('requests.post') as mock_post:
+        mock_response = MagicMock()
+        mock_response.status_code = 201
+        mock_post.return_value = mock_response
+
+        github_permission_manager.post_comment_on_issue(payload, comment_body)
+
+        mock_post.assert_called_once_with(
+            'https://api.github.com/repos/org/repo/issues/1/comments',
+            headers=github_permission_manager.auth_headers,
+            json={"body": comment_body}
+        )
+
+        assert mock_post.called
```

Comment:
the `GithubPermissionsManager` can take a requests module at instantiation so can't we just give it a mock module rather than mocking post

Reply:
Changed this.